### ISSUE TITLE
Valve Index support

### ISSUE THREAD

**Issue by Slydk1982** [KozGit/DOOM-3-BFG-VR#307]
Are there any chance that there will be added full index support?? 
I can play it now on the controller, but you have to give something up for making other things work. Do you want to jump or run?? Will you reload or use flashlight

**Comment by CarlKenner** [comment]
Yes, of course. I don't have a Valve Index controller though, which is a huge problem when trying to develop for it.

Can you be much more specific about all the issues you are experiencing with the index controllers and how you are trying to use them and how things would be implemented better? So that someone who doesn't have one can try to fix it? The controllers aren't properly documented, so it's hard to develop for them without them or assistance from people who have them.

Also, I don't know if the rendering is correct. Apparently, the display panels in the Valve Index are angled and not parallel, but I haven't checked if our rendering engine supports that (Dolphin VR certainly doesn't support that).

**Comment by Slydk1982** [comment]
Okay, I can try :)
The controllers act a bit like the touch controllers. On touch you have Y X B A buttons, but on the index they are B and A on both left and right. The upper button is B and the lower is A.
When I use the A button, it seems the game thinks its the grib button. So A and the grib button do the same thing.
Also on touch controllers you could press left stick to turn your flashlight on and off. I can still do that, but if I move in any direction, the game thinks I push the stick. So im running around and turning my flashlight on and off all the time if I use the left stick.

The render looks very okay. I put my Index to run in 120 Hz and Doom 3 seems to be okay with that. It says 120 Hz in the refresh rate in the options.
The only real problem seems to be the controllers. If the A buttons could get separated from the grip buttons, it could get very playable. And if you can get that left and right stick trigger fixed, it will be awesome :)
The clear display on the index makes Doom 3 look very pretty :)

**Comment by KozGit** [comment]
So, I have an Index & controllers - full support is definitely in the works.
The rendering works fine - although the displays are canted, steam VR defaults to correcting for this automatically via the projection matrices and distortion correction, kind of like enabling parallel projections on the Pimax.
I should have more time to work on this starting in the near future, will update as I make progress. :)

**Comment by Slydk1982** [comment]
Sounds great, man :)

**Comment by AndersMalmgren** [comment]
I get massive headache when playing this on the Index, are you letting steamvr project the image for you? This is the only game that has problem. Unity and unreal games seems to project the image correctly for the Index HMD

**Comment by KozGit** [comment]
The projection should be correct for the index - this uses native steamVR projection.  You may want to go into the settings menu in game to check if it is set to manual IPD and adjust accordingly.

While testing on the index I haven't noticed any issues, so any more information is appreciated.

**Comment by AndersMalmgren** [comment]
It was set to 64 which is pretty close to my physical ipd setting of 65mm. How can I disable manual and let steamvr choose? I still get headaches at 65mm btw

**Comment by booglysnot** [comment]
Issue: moving forward on the Valve Index controllers toggles the flashlight on and off.  Is there a workaround or update for this?

**Comment by Lagahan** [comment]
> 
> 
> Issue: moving forward on the Valve Index controllers toggles the flashlight on and off. Is there a workaround or update for this?

Its caused by valve's workaround for analogs being hard to click near the edge. I just unbound the analog click on the left stick using custom bindings and it went away. Now i just click the trackpad for the flashlight.

~~Edit: I'd also like to default the IPD settings to SteamVR's settings. Gets a bit confusing configuring this stuff.~~

Nevermind it already does

**Comment by redcloud111** [comment]
I am trying to get my Valve index controllers to work with Fully Possessed. When I try to remap the controls in Steam VR devices, it won't let me. This is stopping me from changing a few key things. Does anyone have any steps on how to get this working with index controllers, or should I just wait for the mod to do it? Thanks

**Comment by KozGit** [comment]
Unfortunately, if you can't rebind the control you want through the settings->controls->key bindings  menu ( which is poorly named as it also lets you rebind controllers :) ), you will need to wait for an update.

**Comment by redcloud111** [comment]
Thanks! Do you know how I can clear all bindings from a config file? The problem is that I have some bindings listed that I can't seem to remove. And I have several conflicts. I would like to start from a blank config file for all bindings and add them one by one?

**Comment by Slydk1982** [comment]
So, its been 3 months since I made this post.. Are there any news on a update??

**Comment by jdawgzim** [comment]
@KozGit 
I got an Index too and I'm super excited to play through Doom 3 again on the Index!  Thank you for all your hard work!

**Comment by Slydk1982** [comment]
> @KozGit
> I got an Index too and I'm super excited to play through Doom 3 again on the Index!  Thank you for all your hard work!

Did you get it to work??

**Comment by redcloud111** [comment]
I have been periodically checking here for an update. Since I haven't heard anything, I assume the problem is still there. You might want to try it and see if you can figure it out. I wasn't able to.

**Comment by KozGit** [comment]
Use can try the console commands 'unbind' to unbind something specific, or 'unbindall' to remove all binding and start clean.

**Comment by jdawgzim** [comment]
@Slydk1982  
Oh, I'm waiting for the next update.  I'm currently playing [Brutal Doom VR](https://www.reddit.com/r/virtualreality/comments/cwry4b/brutal_doom_vr_oneclick_installpack_easy_setup_in/) on my Index and it's very 
awesome.

**Comment by redcloud111** [comment]
Koz. Thanks for the console command. I was able to unbindall. Problem is whenever I try to select left, right, forward, backbedall with the right thumbstick, it registers the same binding, which is the A button with the carrot to the left. Also, I am guessing some of the problem is the fact I can't save any custom controller settings in Steam Vr for BFG. So, again, not sure what to do to make it work. If you ever get around to releasing a tweak for the Valve controllers, I will try again then. Thanks!

**Comment by redcloud111** [comment]
Just a head's up that I tried again. I had success. What I did was unbindall key mappings. Then I went through them one at a time. I was also able to save my valve controller settings, which I was not able to do before. I assume this was a Steam problem. Anyway, I am able to make the Index controllers works flawlessly with the mod now. Great work!

**Comment by DinnerBuffet** [comment]
Could someone please post their Valve Index bindings? It's quite a headache trying to come up with something from scratch.

**Comment by NPi2Loup** [comment]
I use this config : 
[D3BFGConfig .cfg](https://github.com/KozGit/DOOM-3-BFG-VR/files/5759518/D3BFGConfig.cfg.txt)

Note that, I made a release with the pending work of the core coder of this mod
Found them here : https://github.com/NPi2Loup/DOOM-3-BFG-VR/releases

I have made some fix to the binding system, it's easier to bind action with Valve Index.

**Comment by DinnerBuffet** [comment]
Thank you! I will try your release and config

**Comment by CanIbiteYou** [comment]
I would like to use the wulfren texture mod with the fully posessed vr mod , how to do that ?

**Comment by NPi2Loup** [comment]
For those who look for this issue.
check #362 @litbeep write a great guide in reddit  for index
